File: app/src/main/java/com/example/files/models/AnalyzerCategory.java
package com.example.files.models;

import android.app.Activity;
import android.content.Context;
import android.text.format.Formatter;

import com.example.files.activities.StorageAnalyzer;

import java.util.ArrayList;

public class AnalyzerCategory extends ArrayList<JFile> {

    StorageAnalyzer.Type TYPE;
    long size = 0;

    public AnalyzerCategory(StorageAnalyzer.Type type) {
        this.TYPE = type;
    }

    @Override
    public boolean add(JFile jFile) {
        this.size += jFile.getSize();
        return super.add(jFile);
    }

    public long getSize() {
        return size;
    }

    public void setSize(long size) {
        this.size += size;
    }

    public String getSize(Context context) {
        return Formatter.formatFileSize(context, size);
    }
}

File: app/src/main/java/com/example/files/models/AnalyzerCategory.kt
package com.example.files.models

import android.content.Context
import android.text.format.Formatter
import com.example.files.activities.StorageAnalyzer

class AnalyzerCategory(val type: StorageAnalyzer.Type) : ArrayList<JFile>() {
    private var mSize: Long = 0

    override fun add(element: JFile): Boolean {
        this.mSize += element.size
        return super.add(element)
    }

    fun setSize(size: Long) {
        this.mSize += size
    }

    fun getSize(): Long {
        return mSize
    }

    fun getSize(context: Context?): String {
        return Formatter.formatFileSize(context, mSize)
    }
}
